### PR TITLE
Add documentation for ByteChunk, IBitmap

### PR DESCRIPTION
**Pull request by QrchackOfficial**
Please do check if the sizes are actually in bytes, I'm doing guesswork by looking at the fact there's memcpy() and the likes used in the implementation

### CODE REVIEW

**Comment by QrchackOfficial** [comment]
Please check if I understand IBitmap correctly before merging

File: IPlug/IGraphics/IGraphicsStructs.h [hunk 1]
```diff
@@ -27,14 +27,29 @@ enum EDirection { kVertical, kHorizontal };
...
   int W;
+  /** Bitmap height (in pixels) */
   int H;
-  /** Number of states (for multibitmaps) */
+  /** Number of frames (for multibitmaps) */
   int N;
+  /** \c True if the frames are positioned horizontally */
   bool mFramesAreHorizontal;
-  double mSourceScale; // i.e. highest res available for this resource
+  /** Maximum scaling allowed for the bitmap (typically 1) */
```
Comment (QrchackOfficial) [inline IPlug/IGraphics/IGraphicsStructs.h]: Please check if this is correct before merging

Comment (olilarkin) [reply IPlug/IGraphics/IGraphicsStructs.h]: Bitmap stuff is being worked on currently, will make sure docs are correct when it is finialised! thanks

**Comment by QrchackOfficial** [comment]
cb7053b1eb460c466447deaecd385a377774e4c8 looks really sweet, especially when you're using Visual Studio for Peek Definition:

image

This results in the following output from Doxygen as well:

image

File: IPlug/IPlugBase.h [hunk 2]
```diff
@@ -119,9 +137,16 @@ class IPlugBase
...
+   * @return Sample rate (in Hz). Defaults to 44100.0
+   */
   double GetSampleRate() const { return mSampleRate; }
   int GetBlockSize() const { return mBlockSize; }
+  /** @return Plugin latency (in samples)
+   * @todo Please check this and remove this note once done
+   */
   int GetLatency() const { return mLatency; }
+  /** @return \c True if the plugin is currently bypassed */
   bool GetIsBypassed() const { return mIsBypassed; }
```
Comment (QrchackOfficial) [inline IPlug/IPlugBase.h]: This is really ugly, can we GetIsBypassed() -> IsBypassed() or GetIsBypassed -> GetBypassed() for IPlug2?

Comment (olilarkin) [reply IPlug/IPlugBase.h]: Yes I’ve been thinking about prefixing every getter with get

Comment (QrchackOfficial) [reply IPlug/IPlugBase.h]: Is- works for telling you it's a bool instantly, but that's your call, either is fine, just need to stick with just one style

Comment (olilarkin) [reply IPlug/IPlugBase.h]: What do you think about things like NParams() NPresets()?

Comment (QrchackOfficial) [reply IPlug/IPlugBase.h]: I think getNumParams() would be way more descriptive

**Comment by QrchackOfficial** [comment]
Also I'd suggest merging (possibly without IPlug/IGraphics/IGraphicsStructs.h as it's subject to change) while the codebase hasn't changed all that much because of documentation, should be way easier to merge than after we think we're done (and suddenly there's tons of conflits to be resolved)

**Comment by olilarkin** [comment]
Are you talking about merging this? I will, just give me a sec

**Comment by olilarkin** [comment]
I’m not going to merge to master yet if that’s what you meant, lots of Apis need to be sorted out before people start using them in anger

**Comment by QrchackOfficial** [comment]
Sure, I meant into iplugquake, just though it'll save tons of conflict resolving later on if this pull request just kind of sits in limbo for a month

File: IPlug/IPlugBase.h (hunk 2 above)
Comment (olilarkin) [reply IPlug/IPlugBase.h]: could you create a separate issue for suggestion such as this? to help keep track

**Comment by olilarkin** [comment]
okay I've merged, with a few changes. I don't think we need to document .cpp files. it should be in the header files. Also it would be great to try and keep the doxygen comments as compact as possible. where something can all go on 1 line

**Pull request merged**